### PR TITLE
discard:provide a predicate to check whether the throwable should be traced

### PR DESCRIPTION
**Pull request by pleasecheckhere2016**
### Describe what this PR does / why we need it
提供了一种可以通过predicate判断该异常是否应该被trace的方法。

### Describe how you did it
```java
Predicate<Throwable> throwablePredicate = new Predicate<Throwable>() {
            @Override
            public boolean test(Throwable throwable) {
                if(throwable instanceof TraceException) {
                    return true;
                } else if (throwable instanceof IgnoreException) {
                    return false;
                }
                return false;
            }
        };
Tracer.setExceptionPredicate(throwablePredicate);
Assert.assertTrue(Tracer.shouldTrace(new TraceException()));
Assert.assertFalse(Tracer.shouldTrace(new IgnoreException()));
```

### CODE REVIEW

**Comment by sczyh30** [comment]
Hi, thanks for contributing. Could you please [sign the CLA here](https://cla-assistant.io/alibaba/Sentinel?pullRequest=1495)? And please make sure the email of your commits match your GitHub email. You may refer to the instruction [here](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user), or rebase your commits with the correct email.

---

感谢贡献，请将 commit 对应的 email 调整成与 GitHub 的 email 相匹配并 [确认一下 CLA](https://cla-assistant.io/alibaba/Sentinel?pullRequest=1495)。

File: sentinel-core/src/main/java/com/alibaba/csp/sentinel/Tracer.java [hunk 1]
```diff
@@ -181,6 +202,10 @@ protected static boolean shouldTrace(Throwable t) {
         if (t == null || t instanceof BlockException) {
             return false;
         }
+        if(exceptionPredicate != null) {
```
Comment (sczyh30) [inline sentinel-core/src/main/java/com/alibaba/csp/sentinel/Tracer.java:205]: Could you please reformat your code here? Like:

```java
if (exceptionPredicate != null) {
```

File: sentinel-core/src/test/java/com/alibaba/csp/sentinel/TracerTest.java [hunk 2]
```diff
@@ -52,6 +53,25 @@ public void setExceptionsToTrace() {
         Assert.assertFalse(Tracer.shouldTrace(new Exception()));
     }
 
+    @Test
+    public void setExceptionPredicate() {
+
+        Predicate<Throwable> throwablePredicate = new Predicate<Throwable>() {
+            @Override
+            public boolean test(Throwable throwable) {
+                if(throwable instanceof TraceException) {
```
Comment (sczyh30) [inline sentinel-core/src/test/java/com/alibaba/csp/sentinel/TracerTest.java:62]: Ditto

**Comment by pleasecheckhere2016** [comment]
> Hi, thanks for contributing. Could you please [sign the CLA here](https://cla-assistant.io/alibaba/Sentinel?pullRequest=1495)? And please make sure the email of your commits match your GitHub email. You may refer to the instruction [here](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user), or rebase your commits with the correct email.
> 
> 感谢贡献，请将 commit 对应的 email 调整成与 GitHub 的 email 相匹配并 [确认一下 CLA](https://cla-assistant.io/alibaba/Sentinel?pullRequest=1495)。

Hi,可以帮忙把本次PR关闭吗，我需要修改一下email和用户名重新提交

**Pull request closed without merging**